servers/rendering/renderer_geometry_instance.cpp

```diff
@@ -75,7 +75,8 @@ void RenderGeometryInstanceBase::set_transform(const Transform3D &p_transform, c
 
 	float max_scale = MAX(model_scale_vec.x, MAX(model_scale_vec.y, model_scale_vec.z));
 	float min_scale = MIN(model_scale_vec.x, MIN(model_scale_vec.y, model_scale_vec.z));
-	non_uniform_scale = max_scale >= 0.0 && (min_scale / max_scale) < 0.9;
+	const float epsilon = 0.0000001f;
+	non_uniform_scale = max_scale >= 0.0 && (min_scale / (max_scale + epsilon)) < 0.9;
```

Comment:
This will change the behavior from treating scale of `(0, 0, 0)` as uniform to treating it as non-uniform, unsure if that's desired (as `0/0` is `NaN` which is not less than 0.9)

Reply:
0/0 in C++ is  not defined and leads to undefined behavior. 

In C++, the comparison of a NaN value with any other float value using regular comparison operators (==, !=, <, <=, >, >=) results in false. This behavior is due to the IEEE 754 standard for floating-point arithmetic, which C++ adheres to.

NaN has no algorithmic purpose in scale AFAIK. Actually (min_scale / max_scale) should not be even evaluated if max_scale < 0, though it will be evaluated with max_scale = 0 and it will emit a runtime_error exception. And yes however small is the epsilon is, it will change the scale slightly. The result for (min_scale / (max_scale + epsilon)) however is guaranteed to be less then1.0,  if max_scale > min scale. But it is not guaranteed to be < 0.9. 

I can't judge the 0.9 value here. There are other possible solutions the obvious one is to evaluate:

```
non_uniform_scale = max_scale > 0.0 && (min_scale / max_scale) < 0.9;
```
This will return false if max_scale is == 0.0, and no one need the epsilon in this case.
Other solution if you allow max_scale to be 0, check if min_scale is less then max_scale and still the max_scale > 0.0 or (max_scale + epsilon) is > 0.0 (which will turn into 0 if scale is -0.0000001f).

Idk what are the appropriate intended border values for this expression, but I would not leave the runtime_error exception handle this ...

Reply:
I know these things thank you

But the fact remains that this changes the behavior for a scale of `(0, 0, 0)` which needs to be decided on, but changing to what you suggest does preserve the behavior

Reply:
Reply: 

> It has the well defined behavior of returning NaN

No it is not. It might be in some other languages but not in C or C++ https://www.open-std.org/jtc1/sc22/wg21/docs/papers/2017/n4713.pdf

8.1 clause 4:

4 If during the evaluation of an expression, the result is not mathematically defined or not in the range of
representable values for its type, the behavior is undefined. [ Note: Treatment of division by zero, forming a
remainder using a zero divisor, and all floating-point exceptions vary among machines, and is sometimes
adjustable by a library function. — end note

Reply:
I tested it in `clang++` and `msvc` with `c++17` and it works correctly, returning `false` for this check when both are zero, this is the standard generally and matches the IEEE standard